### PR TITLE
Add webnn slice tests

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @Honry @huningxin @dontcallmedom PTAL, thanks.

### CODE REVIEW

File: webnn/resources/test_data/slice.json [hunk 1]
```diff
@@ -0,0 +1,772 @@
...
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [2, 2],
+      "sizes": [2, 4],
```
Comment (fdwr) [inline webnn/resources/test_data/slice.json:94]: It's interesting that WebNN chose `starts` and `sizes`, whereas most other frameworks I'm familiar with chose `starts` and `ends`, which makes it easier to specify relative endpoints (e.g. from element 0 to element N-1). Shrug. (no action expected)

File: webnn/resources/test_data/slice.json [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [1, 0, 2, 1, 0],
+      "sizes": [-1, 2, -1, 1, -1], // [-1, 2, -1, 1, -1] is equal to [1, 2, 1, 1, 1]
```
Comment (fdwr) [inline webnn/resources/test_data/slice.json:430]: Is -2 a valid size? Like I said above, every framework I know of has the ability to accept negative end values, so you can easily crop off the last N elements, but WebNN does not appear to have that, requiring the caller to instead resolve the size beforehand, and yet the start coordinate does accept negative values which are relative, and the inconsistency here doesn't make sense 🤔.

**Review by fdwr — APPROVED** [review]

File: webnn/resources/test_data/slice.json (hunk 1 above)
Comment (BruceDai) [reply webnn/resources/test_data/slice.json:94]: Yes. Some use `ends`/`stop` like ONNX ([Slice](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Slice)) , OpenVINO ([Slice](https://docs.openvino.ai/latest/openvino_docs_ops_movement_Slice_8.html)), some use `sizes`/`size`, like DML ([SLICE](https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_slice_operator_desc)),  TensorFlow ([slice](https://js.tensorflow.org/api/latest/#slice)).

Comment (fdwr) [reply webnn/resources/test_data/slice.json:94]: Plus:
- NumPy begin,end (https://numpy.org/doc/stable/user/basics.indexing.html?highlight=slice#slicing-and-striding
- PyTorch start,size (narrow)(https://pytorch.org/docs/stable/generated/torch.narrow.html#torch.narrow)

**Comment by BruceDai** [comment]
@Honry All checks passed, PTAL, thanks.

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**